lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+        text=True
+    )
+    return (str(result.stdout).strip().split('-'))[0]
+
+def display_version():
+    version = get_version()
+
+    parser = argparse.ArgumentParser(description="Current version of LCLS Naming Tool")
+    parser.add_argument('-v', '--version', action='store_true')
+    args = parser.parse_args()
+    
+    if args.version:
+        print(version)
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('taxons/functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('taxons/fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('taxons/ccc_taxon.json') as ccc_file:
+        ccc_dict = json.load(ccc_file)
+
+
+def functional_component_is_valid(fc_taxon):
+
+    fc_length = len(fc_taxon)
+    
+    if (4 < len(fc_taxon) < 9):
+        fc_prefix = fc_taxon[:2]  # max length is currently 2 letters
+        fc_seq_num = fc_taxon[2:fc_length-2]  # max length set to 4
+        fc_source_ltr = fc_taxon[fc_length-2:fc_length-1] # max length is currently 1 letter
+        fc_beam_num = fc_taxon[fc_length-1:fc_length]     # max length is currently 1 digit
+    else:
+        return False
+
+    # validate FFFFF
+    try:
+        fc_dict[fc_prefix]
+        beam_sources[fc_source_ltr]
+        beam_numbers[fc_beam_num]
+
+        try:
+            int(fc_seq_num)
+
+            try:
+                1 / int(fc_seq_num[0])  # zero padding not allowed
+            except 0:
+                return False
+
+        except:
+            return False
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def fungible_is_valid(fg_taxon):
+    try:
+        fg_elements = fg_taxon.replace(
+            '_', ' ').split()
+
+        for name in fg_elements:
+            fg_dict[name]
+
+    except KeyError:
+        return False
+
+    else:
+        return True
+
+
+def constituent_component_is_valid(ccc_taxon):
+    try:
+        ccc_dict[ccc_taxon]
+    except:
+        return False
+    else:
+        return True
+
+
+def increment_is_valid(nn_taxon):
+    # check the increment is an int and has at least 2 digits
+    if len(nn_taxon) >= 2:
+        try:
+            int(nn_taxon)
+        except 0:
+            return False
+
+        else:
+            return True
+    else:
+        return False
+
+
+def starts_alphanumeric(taxon):
+    # check the element starts with a letter or number 
+    match = re.search('^[a-zA-Z0-9]', taxon)
+    if match:
+        return True
+    else:
+        return False
+
+
+def validate(user_input):
+    if ('.' in user_input) or (len(user_input) > 60):
+        return False
+
+    name = [x.strip() for x in user_input.upper().split(':')]
+
+    for element in name:
+        if not starts_alphanumeric(element):
+            print('Invalid')
+            return False
```

Comment:
Some small observation: If we always do the same print regardless of the context when we return True or False, maybe we should implement this in main instead:
```
if validate(line):
    print("Valid")
else:
    print("Invalid")
```